From 9fd5d5a35acabae4628fe0e5f70a0a2d5b990ed0 Mon Sep 17 00:00:00 2001
From: "Thulasimani,Sivakumar" <sivakumar.thulasimani@intel.com>
Date: Wed, 23 Dec 2015 15:03:30 +0530
Subject: [PATCH 06/19] FOR_UPSTREAM [VPG]: drm/i915: Disable MaxFifo during
 modeset

Maxfifo should be enabled only when one plane is present.
This is handled during the flip path alone but not in the
modeset path, where during modeset a new plane can be
enabled. This patch disables maxfifo when we begin
modeset.

Change-Id: I02a5a5603a9789d3aea1c0a8b433e8bd302255a8
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-30099
Signed-off-by: Sivakumar Thulasimani <sivakumar.thulasimani@intel.com>
Reviewed-on: https://android.intel.com:443/498181
---
 drivers/gpu/drm/i915/intel_display.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/gpu/drm/i915/intel_display.c b/drivers/gpu/drm/i915/intel_display.c
index 3f5ca99..2b89fee 100644
--- a/drivers/gpu/drm/i915/intel_display.c
+++ b/drivers/gpu/drm/i915/intel_display.c
@@ -12645,6 +12645,7 @@ static int __intel_set_mode(struct drm_crtc *crtc,
 
 	*saved_mode = crtc->mode;
 
+	intel_update_maxfifo(dev_priv, crtc, false);
 	/* Hack: Because we don't (yet) support global modeset on multiple
 	 * crtcs, we don't keep track of the new mode for more than one crtc.
 	 * Hence simply check whether any bit is set in modeset_pipes in all the
-- 
1.9.1

